compiler/rustc_lint/src/builtin.rs

```diff
@@ -896,7 +896,7 @@ declare_lint! {
     /// [issue #41686]: https://github.com/rust-lang/rust/issues/41686
     /// [`cargo fix`]: https://doc.rust-lang.org/cargo/commands/cargo-fix.html
     pub ANONYMOUS_PARAMETERS,
-    Allow,
+    Warn,
```

Comment:
If you move forward with this, I think it would be good to update the documentation just a few lines above this that discusses why it is "allow"ed (the whole paragraph will probably need to be rewritten).